classes/form/username_form.php

```diff
@@ -95,4 +105,31 @@ public static function active_user_exists($username) {
         global $DB;
         return $DB->record_exists('user', array('username' => $username, 'deleted' => 0, 'suspended' => 0));
     }
+
+    /**
+     * Confirms whether the given email address exists (and neither deleted, nor suspended).
+     *
+     * @param $username
+     * @return bool
+     * @throws \dml_exception
+     */
+    public static function active_user_by_email_exists($username) {
+        global $DB;
+        return $DB->record_exists('user', array('email' => $username, 'deleted' => 0, 'suspended' => 0));
+    }
+
+    /**
+     * Returns true if finds the email more than once.
+     *
+     * @param $username
+     * @return boolean
+     */
+    public static function is_email_duplicate($email) {
+        global $DB;
+        if (sizeof($DB->get_fieldset_select('user', 'email', "email = '{$email}'")) > 1) {
```

Comment:
DB has some specific functions to count the number of records (count_records(), count_records_select(), count_records_sql()), which would be more suitable here.